packages/grid/x-data-grid/src/hooks/features/columnHeaders/useGridColumnHeaders.tsx

```diff
@@ -109,6 +107,11 @@ export const useGridColumnHeaders = (props: UseGridColumnHeadersProps) => {
   const prevScrollLeft = React.useRef(0);
   const currentPage = useGridVisibleRows(apiRef, rootProps);
 
+  const derivedHeaderHeight = React.useMemo(
```

Comment:
I was thinking about having it simply as `headerHeight` but `derivedHeaderHeight` (same for the row height) sounds more descriptive because it's calculated on the fly. I don't have strong preferences for the naming of the constant so if you still want I can change it.